File: frontend/multilang_examples/hello.hs
greet :: String -> String
greet name = "Hello, " ++ name ++ "!"
add :: Int -> Int -> Int
add a b = a + b
main :: IO ()
main = do
    putStrLn (greet "World")
    putStrLn ("Sum: " ++ show (add 3 5))

File: frontend/multilang_examples/hello.cpp
#include <iostream>
#include <string>
std::string greet(const std::string& name) {
    return "Hello, " + name + "!";
}
int add(int a, int b) {
    return a + b;
}
int main() {
    std::cout << greet("World") << std::endl;
    std::cout << "Sum: " << add(3, 5) << std::endl;
    return 0;
}
